utils/docker/Dockerfile.el.9

```diff
@@ -0,0 +1,206 @@
+# Copyright 2022 Intel Corporation
+# All rights reserved.
+#
+# 'recipe' for Docker to build an image of EL 9 based
+# environment for building the DAOS project.
+#
+
+# Pull base image
+ARG BASE_DISTRO=almalinux:9
+FROM $BASE_DISTRO
+LABEL maintainer="daos@daos.groups.io"
+
+# Intermittent cache-bust.  Used to reduce load on the actual CACHEBUST later.
+ARG CB0
+
+# Use local repo server if present
+ARG REPO_FILE_URL
+RUN set -e;                                                      \
+    if [ -n "$REPO_FILE_URL" ]; then                             \
+        cd /etc/yum.repos.d/ &&                                  \
+        curl -k -f -o daos_ci-el9-artifactory.repo.tmp           \
+             "$REPO_FILE_URL"daos_ci-el9-artifactory.repo &&     \
+        for file in *.repo; do                                   \
+            true > $file;                                        \
+        done;                                                    \
+        mv daos_ci-el9-artifactory.repo{.tmp,};                  \
+    fi;                                                          \
+    dnf --assumeyes install epel-release dnf-plugins-core;       \
```

Comment:
You need the el8 changes from https://github.com/daos-stack/daos/pull/9142

There's an interesting thing with epel-release in that the package exists in CentOS, and that adds the repos but the package also exists in epel itself and that can also change the configuration so you need to install it twice here.

Without the second install then it was included in the "dnf upgrade" on L64 and as it was reconfiguring yum it was installed first and caused yum to do two passes which caused a bad interaction between yum and overlayfs which lead to deadlock.  I believe yum is patched in EL8 (and therefore 9) to close/reopen the file for the second pass but it would still be better to do a second install here.